nexus/types/src/deployment.rs

```diff
@@ -97,6 +101,47 @@ impl SledResources {
     }
 }
 
+/// Policy and database inputs to the Reconfigurator planner
```

Comment:
(sorry, wasn't sure where to hang this comment)

In #5345 I added a common function for assembling `UnstableReconfiguratorState`.  This is used in omdb plus one of the reconfigurator-cli tests.  It's closely related but not quite the same as what Nexus assembles for planning.  I'm wondering if it would make sense to:

- have the common function assemble a `PlanningInput`
- `impl From<PlanningInput> for UnstableReconfiguratorState` -- maybe the `UnstableReconfiguratorState` is just an object with the contents of `PlanningInput` that impls `Serialize`/`Deserialize`?

Again, not a blocker for this PR.